src/Homepage.js

```diff
@@ -1,11 +1,29 @@
 import { useState, useEffect } from "react";
 
+let readSpeech = false;
+let speech = new SpeechSynthesisUtterance();
+speech.lang = "en";
```

Comment:
Not good, you need to have these fields inside the component, not here.
useState for the readSpeech boolean
and useRef for the SpeechSynthesisUtterance, and language, you need to make sure you initialize these in useEffect that happens once